src/v3/src/transformernew/attribute/transform.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IdxTransaction } from '@okta/okta-auth-js';
+import { 
+  AutoCompleteValue, 
+  InputAttributes, 
+  Nullable,
+  FormBag,
+  FieldElement,
+} from '../../types';
+
+const autocompleteValueMap = new Map<string, AutoCompleteValue>([
+  ['identifier', 'username'],
+  ['password', 'current-password'],
+  ['newPassword', 'current-password'],
+  ['passcode', 'one-time-code'],
+  ['totp', 'one-time-code'],
+  ['phoneNumber', 'tel-national'],
+  ['firstName', 'given-name'],
+  ['lastName', 'family-name'],
+  ['email', 'email'],
+]);
+
+const getKeyFromMap = (
+  map: Map<string, AutoCompleteValue>,
+  inputName: string,
+): string | undefined => {
+  let isMatch;
+  map.forEach((value: AutoCompleteValue, key: string) => {
+    if (inputName.match(key)?.length) {
+      isMatch = key;
+    }
+  });
+  return isMatch;
+};
```

Comment:
Reverse lookup? Why is this needed?

Reply:
@glenfannin-okta ^

Reply:
So, this was done because the keys that are passed to this function are flattened into a string. 

So for example phoneNumber will be presented as `authenticator.phoneNumber` or password will be presented as `credentials.passcode`, otp will either come as `credentials.passcode` or `totp`, etc. 

However, I don't know of every format phoneNumber can come, because it is present when registering as well as when authenticating as well as when setting up a new phone for an authenticator. 

Because of these variables, I just rely on the last part of the field name. So to do that, I have to perform a match on the key of the map (which is what this function does). 

If you are confident the field names are standardized and there are no variances then we can get rid of it but I was not confident in making that assumption which is why I created this lookup function.